pkg/ovncontroller/daemonset.go

```diff
@@ -150,6 +152,19 @@ func CreateOVNDaemonSet(
 		daemonset.Spec.Template.Spec.NodeSelector = *instance.Spec.NodeSelector
 	}
 
+	if topology != nil {
```

Comment:
@olliewalsh The original `DaemonSet` code does not set anything except `NodeSelector`. It makes sense in this case. and I'm wondering if we prefer to not have topology in this case. In theory it doesn't hurt and `TopologyConstraints` gets ignored, but `NodeAffinity` might be processed [1] (and it's technically part of the `Topology` interface).
I could simplify the code further and remove `TopologySpreadConstraints` because it does not make sense here, but keep affinity because of [1].
I'm going to do this update, but let me know if you have other points we should consider.
 
[1] https://github.com/kubernetes/kubernetes/blob/master/pkg/controller/daemon/util/daemonset_util.go#L226